### PR TITLE
Add state cve report

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/176449977

### CODE REVIEW

File: cmd/state/internal/cmdtree/cve.go [hunk 1]
```diff
@@ -25,3 +26,29 @@ func newCveCommand(prime *primer.Values) *captain.Command {
...
+	return captain.NewCommand(
+		"report",
+		locale.Tl("cve_report_title", "Vulnerability Report"),
+		locale.Tl("cve_report_cmd_description", "Show a detailed report of project vulnerabilities"),
+		prime.Output(),
+		prime.Config(),
+		[]*captain.Flag{},
+		[]*captain.Argument{
+			{
+				Name:        locale.Tl("cve_report_namespace_arg", "organization/project"),
```
Comment (MDrakos) [inline cmd/state/internal/cmdtree/cve.go:45]:
```suggestion
				Name:        locale.Tl("cve_report_namespace_arg", "Organization/Project"),
```

File: internal/runners/cve/cve.go [hunk 2]
```diff
@@ -120,6 +120,14 @@ type SeverityCountOutput struct {
 	Severity string `locale:"severity,Severity"`
 }
 
+func (od *outputDataPrinter) printFooter() {
+	od.output.Print("")
+	od.output.Print([]string{
+		locale.Tl("cve_hint_report", "To view a detailed report for this runtime, run [ACTIONABLE]state cve report[/RESET]"),
+		locale.Tl("cve_hint_specific_report", "For a specific runtime, run [ACTIONABLE]state cve report [orgName/projectName][/RESET]"),
```
Comment (MDrakos) [inline internal/runners/cve/cve.go:127]:
```suggestion
		locale.Tl("cve_hint_specific_report", "For a specific runtime, run [ACTIONABLE]state cve report [Organization/Project][/RESET]"),
```

File: internal/runners/cve/report.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+			}
+			rd.output.Print(fmt.Sprintf("  %s %-10s [ACTIONABLE]%s[/RESET]", bar, severity, d.CveId))
+		}
+		rd.output.Print("")
+	}
+
+	rd.output.Print("")
+	rd.output.Print([]string{
+		locale.Tl("cve_report_hint_cve", "To view a specific CVE, run [ACTIONABLE]state cve open [cve-id][/RESET]."),
+	})
```
Comment (MDrakos) [inline internal/runners/cve/report.go:205]: Wrapping this in a `[]string` will prepend a ` - ` to it. Is this the desired output here?

File: test/integration/cve_int_test.go [hunk 4]
```diff
@@ -26,16 +26,38 @@ func (suite *CveIntegrationTestSuite) TestCveSummary() {
...
 
+func (suite *CveIntegrationTestSuite) TestCveReport() {
+	suite.OnlyRunForTags(tagsuite.Cve)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	ts.LoginAsPersistentUser()
+
+	cp := ts.Spawn("cve", "report", "ActiveState/ActivePython-3.7")
```
Comment (MDrakos) [inline test/integration/cve_int_test.go:47]: This project isn't owned by us, and could be updated to patch these vulnerabilities. You may want to fork this project, or create your own so this test does not end up randomly breaking

**Review by MDrakos — APPROVED** [review]
Makes sense. My comments are just nits and suggestions.

File: internal/runners/cve/report.go (hunk 3 above)
Comment (mdrohmann) [reply internal/runners/cve/report.go:205]: Yes, in the mock these are also presented as a list.

File: cmd/state/internal/cmdtree/cve.go [hunk 5]
```diff
@@ -25,3 +26,29 @@ func newCveCommand(prime *primer.Values) *captain.Command {
 	cmd.SetGroup(PlatformGroup)
 	return cmd
 }
+
+func newReportCommand(prime *primer.Values) *captain.Command {
+	report := cve.NewReport(prime)
+	params := cve.ReportParams{
+		Namespace: &project.Namespaced{},
```
Comment (Naatan) [inline cmd/state/internal/cmdtree/cve.go:33]: Why initialize an empty namespace? Shouldn't this be nil if it wasn't provided?

File: internal/runners/cve/report.go [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+			}
+			cves[ve.Version] = append(cves[ve.Version], ve)
+		}
+
+		for ver, vuls := range cves {
+			packageVulnerabilities = append(packageVulnerabilities, DetailedByPackageOutput{
+				v.Name, ver, vuls,
+			})
+		}
+	}
```
Comment (Naatan) [inline internal/runners/cve/report.go:107]: Can we move this into the model?

File: internal/runners/cve/report.go [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+		}
+
+		// Remove this block with story https://www.pivotaltracker.com/story/show/176508772
+		// filter double entries
+		if _, ok := visited[v.Name]; ok {
+			continue
+		}
+		visited[v.Name] = struct{}{}
+
+		cves := make(map[string][]medmodel.Vulnerability)
```
Comment (Naatan) [inline internal/runners/cve/report.go:94]: If we're going to do post-processing on these types at least alias the type.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: cmd/state/internal/cmdtree/cve.go (hunk 5 above)
Comment (mdrohmann) [reply cmd/state/internal/cmdtree/cve.go:33]: This is how we did in [other places](https://github.com/ActiveState/cli/blob/master/cmd%2Fstate%2Finternal%2Fcmdtree%2Factivate.go#L19-L20) to set the namespace as an argument, by calling its `Set()` method as required by the `ArgMarshaler` interface.

**Review by Naatan — APPROVED** [review]

**Pull request merged**